src/gui/MainWindow.cpp

```diff
@@ -379,16 +379,16 @@ void MainWindow::finalize()
 	m_toolsMenu = new QMenu( this );
 	for( const Plugin::Descriptor* desc : getPluginFactory()->descriptors(Plugin::Tool) )
 	{
-		m_toolsMenu->addAction( desc->logo->pixmap(), desc->displayName );
-		m_tools.push_back( ToolPlugin::instantiate( desc->name, /*this*/nullptr )
-						   ->createView(this) );
-	}
-	if( !m_toolsMenu->isEmpty() )
-	{
-		menuBar()->addMenu( m_toolsMenu )->setText( tr( "&Tools" ) );
-		connect( m_toolsMenu, SIGNAL(triggered(QAction*)),
-					this, SLOT(showTool(QAction*)));
+		auto objectName = QString("ToolsMenuAction") + desc->displayName;
+		m_tools[objectName] = ToolPlugin::instantiate(desc->name, nullptr)->createView(this);
+
+		//GUI
+		auto* action = m_toolsMenu->addAction( desc->logo->pixmap(), desc->displayName );
```

Comment:
Here, too, let's use `auto*` for pointers.

Reply:
This is rather subjective since some people do not feel the need to differentiate between raw pointers for ``auto*`` and ``auto``. Feel free to use ``auto*`` however, I don't really have any strong reasons for using either one :+1:.

Reply:
variable can not be declared `const` because we call its modifying member `setObjectName()` after this initialization.

Reply:
> variable can not be declared `const` because we call its modifying member `setObjectName()` after this initialization.

My mistake, no need to include ``const`` then. I forgot to check if some of the functions were modifying.